bpfd/src/main.rs

```diff
@@ -11,7 +11,11 @@ use nix::{
 };
 use systemd_journal_logger::{connected_to_journal, init_with_extra_fields};
 
-#[tokio::main]
+const DEFAULT_BPFD_CONFIG_PATH: &str = "/etc/bpfd/bpfd.toml";
+const DEFAULT_BPFD_STATIC_PROGRAM_DIR: &str = "/etc/bpfd/programs.d";
+const BPFD_ENV_LOG_LEVEL: &str = "RUST_LOG";
+
+#[tokio::main(flavor = "multi_thread", worker_threads = 1)]
```

Comment:
No :smile: 

Main thread = BPF thread.
Spawned thread = RPC thread.

The `current_thread` executor would use a single OS thread to run BOTH those tasks. Not what we want.

The `multi_thread` executor spawns a thread per CPU and schedules work across them. This is cool, because the `RPC` thread will quite likely call `spawn` also, BUT, those workers inherit the capabilities of the main thread unless we drop them.

The correct option is probably to manually configure the runtime using https://docs.rs/tokio/1.20.1/tokio/runtime/struct.Builder.html and use `on_thread_start` to drop permissions

Reply:
Fixed